### ISSUE TITLE
Upgrade Google Analytics API from v3 to v4

### ISSUE THREAD

**Issue by clawdaddy** [metabase/metabase#12905]
**Is your feature request related to a problem? Please describe.**
Google Analytics data currently lacks support for data manipulation. I'd like to be able to do simple manipulation, such as dividing the number of new users by the total number of users on any given day. 
API
https://developers.google.com/analytics/devguides/reporting/core/v3/
Metabase implementation
https://github.com/metabase/metabase/tree/aced697af9c7bc71d95427a7a995fd6655bc8ea0/modules/drivers/googleanalytics/src/metabase/driver/googleanalytics

**Describe the solution you'd like**
Upgrading the API integration to v4 should make some of these features available. 
https://developers.google.com/analytics/devguides/reporting/core/v4#top_of_page

**Describe alternatives you've considered**
I'm not sure. If there was some way to take the raw data obtained from GA and then manipulate it in metabase's code, I suppose that could work as well. That doesn't seem to follow the paradigm metabase utilizes for db queries, though. 

**How important is this feature to you?**
I'd say fairly important. I'm trying to determine the virality of our product, which is best obtained by taking the number of new users and comparing it to some measure of old users. I can obtain each of these values separately, but combining them into a chart doesn't seem possible at this time. 

**Additional context**

**Comment by manuelef** [comment]
@clawdaddy I had the same problems. Do you have any idea if this feature will be prioritized? Thanks

**Comment by camilatweber** [comment]
@clawdaddy I had this problems. I need to use the expression too.
And use new features that are available in v4, like user retention.

**Comment by RobertHeim** [comment]
any news on this?

**Comment by alekseipico** [comment]
Hello 
Have a question as well - I added GA source to Metabase and it shows only [Legacy] tables but no any GA Reporting API V4 [Properties] . 

1) Is this expected behavior ? 

2) I found confusing that message:
https://discourse.metabase.com/t/google-analytics-native-query-documentation/5387
Looks like it's possible to use "Native Query syntax"

Is it true - can I use "Native Query syntax" for requesting V4. I tried but got nothing

**Comment by blastoma** [comment]
+1

**Comment by ashit97** [comment]
👍

**Comment by lpadula** [comment]
Hi, GA3 will be discontinued on 2023: https://support.google.com/analytics/answer/11583528

GA4 support is very important.

**Comment by AndreSchwarzer** [comment]
Seems not to be on the [Roadmap](https://www.metabase.com/roadmap/) for the near future, yet.

**Comment by MrChadMWood** [comment]
Hello! Also looking for an update. Current functionality will no longer work soon as Google is depreciating Universal Analytics. GA4 is already required for newly made properties in Google Analytics. Metabase support for GA4 is critical to any user who uses Metabase for GA, as they will be left with no options after July 2023.

Any update?

**Comment by paoliniluis** [comment]
@MrChadMWood our advise would be to use bigquery to query GA v4 in the meantime

**Comment by likeshumidity** [comment]
GA3 EOL scheduled for 2022-07-01, so will need to be addressed prior to that date.

Many GA3 customers are in flight with conversions, so I would expect need for this to increase as we near June 2022.

**Comment by mikemaring** [comment]
BigQuery is a workaround, but Google Analytics is pretty huge and Universal Analytics is about to be fully deprecated. Are there zero plans to support Google Analytics going forward?

**Comment by paoliniluis** [comment]
@mikemaring zero, supporting ga4 would require to rewrite the entire driver, it’s like starting from scratch

**Comment by saywebsolutions** [comment]
Will it be supported sometime in the future (ie. is it on the roadmap)?

**Comment by paoliniluis** [comment]
@saywebsolutions no, we’re deprecating all Google analytics support. I would suggest you check out how to send your GA data to bigquery which is supported

**Comment by saywebsolutions** [comment]
Thank you
